### PR TITLE
Mm export page fix

### PR DESCRIPTION
**Pull request by mXaln**

### CODE REVIEW

File: src/js/pages/export/ExportPage.js [hunk 1]
```diff
@@ -31,7 +33,19 @@ export class ExportPage extends Component {
     this.props.resetSelected();
   }
 
-  toggleCheck = () => { this.setState({checked: !this.state.checked});}
+  toggleCheck = () => { 
+    this.setState({
+      checked: !this.state.checked,
+      checkedAll: false
+    });
+  }
```
Comment (natibekele) [inline src/js/pages/export/ExportPage.js]: when toggling states, this seems to be the accepted way to do it in react
``` this.setState( prevState => ({checked: !prevState.checked, checkedAll: false})```

File: src/js/pages/export/ExportPage.js [hunk 2]
```diff
@@ -31,7 +33,19 @@ export class ExportPage extends Component {
...
+      checked: !this.state.checked,
+      checkedAll: false
+    });
+  }
+
+  toggleCheckAll = () => { 
+    this.setState({
+      checkedAll: !this.state.checkedAll,
+      checked: false
+    });
```
Comment (natibekele) [inline src/js/pages/export/ExportPage.js]: `this.setState( prevState => ({checkedAll: !prevState.checkedAll, checked: false})`

File: src/js/pages/export/ExportPage.js (hunk 1 above)
Comment (mXaln) [reply src/js/pages/export/ExportPage.js]: @natibekele oh that looks nice, I'll update the code

File: src/js/pages/export/components/ExportCard.js [hunk 3]
```diff
@@ -11,16 +11,14 @@ export class ExportCard extends Component {
...
-      if (nextProps.completedSelected && nextProps.published) {                // button selected and the chapter is published, mark them as selected
+    if ( nextProps.completedSelected !== this.props.completedSelected || nextProps.allSelected !== this.props.allSelected) { //check if the state of the button select completed changed
+      if ((nextProps.completedSelected && nextProps.published) || nextProps.allSelected) {                // button selected and the chapter is published, mark them as selected
         this.setState({checked: true});
         if (!this.state.checked) {             // to avoid adding them twice when they already selected
           nextProps.selections(nextProps.id, true);         // send id and number to save them into an array, true is used for saved, false for remove
         }
       }
-      if (!nextProps.completedSelected && nextProps.published) {   // once we check the checkbox
+      if ((!nextProps.completedSelected || (nextProps.completedSelected && !nextProps.published)) && !nextProps.allSelected) {   // once we check the checkbox
```
Comment (natibekele) [inline src/js/pages/export/components/ExportCard.js]: this whole componentWillReceiveProps is not very easy to read. maybe we could change the name of the completedSelected prop to something like `checked` for instance. it made more sense for me when i substituted completedSelected for checked, that might have only been for me, but in general i think we can try and make this easier to follow

Comment (mXaln) [reply src/js/pages/export/components/ExportCard.js]: @natibekele and change allSelected to checkedAll?

**Pull request merged**